src/rgw/rgw_data_sync.cc

```diff
@@ -706,6 +706,22 @@ static string full_data_sync_index_shard_oid(const string& source_zone, int shar
   return string(buf);
 }
 
+struct read_metadata_list {
+  string marker;
+  bool truncated;
+  list<string> keys;
+  int max_entries;
+
+  read_metadata_list() : truncated(false), max_entries(0) {}
+
+  void decode_json(JSONObj *obj) {
+    JSONDecoder::decode_json("marker", marker, obj);
+    JSONDecoder::decode_json("truncated", truncated, obj);
+    JSONDecoder::decode_json("keys", keys, obj);
+    JSONDecoder::decode_json("max-entries", max_entries, obj);
```

Comment:
"max-entries" isn't part of the json response from RGWOp_Metadata_List. it does have a "count", but i don't think count is useful in RGWListBucketIndexesCR (you can just call keys.size() instead). so you can either drop max-entries from this struct, or replace it with "count" - up to you

Reply:
Replaced with 'count'